### PR TITLE
[com_fields] Fix/Improve checks for empty output in field groups

### PR DESCRIPTION
**Pull request by AndySDH**
### Summary of Changes

This PR improves checks for empty output when rendering a field as part of field groups. Some checks were already there but not thorough, and empty output could occur in rare cases.

This makes sure that a field that has an empty value or an empty output does not get displayed in all cases.

One check comes from the original subfields PR (https://github.com/joomla/joomla-cms/pull/22446/files#diff-02a314737553eb71a8516e925b220cce) that was postponed to Joomla 4.
Another extra check I added myself.

### Testing Instructions

Not much needed really. Just double checks.

### Expected result

Field with empty output should never display

### Actual result

Field with empty output could be displayed in rare occurrences

### Documentation Changes Required

None

### CODE REVIEW

**Review by laoneo — APPROVED** [review]

File: components/com_fields/layouts/fields/render.php [hunk 1]
```diff
@@ -54,14 +54,19 @@
...
-	if (!isset($field->value) || $field->value == '')
+	if (!isset($field->value) || trim($field->value) == '')
 	{
 		continue;
 	}
 
 	$class = $field->params->get('render_class');
 	$layout = $field->params->get('layout', 'render');
 	$content = FieldsHelper::render($context, 'field.' . $layout, array('field' => $field));
+
```
Comment (Quy) [inline components/com_fields/layouts/fields/render.php]: Remove tab.

File: components/com_fields/layouts/fields/render.php [hunk 2]
```diff
@@ -54,14 +54,19 @@
...
 	{
 		continue;
 	}
 
 	$class = $field->params->get('render_class');
 	$layout = $field->params->get('layout', 'render');
 	$content = FieldsHelper::render($context, 'field.' . $layout, array('field' => $field));
+	
+	// If the content is empty do nothing
+	if (!isset($content) || trim($content) == '') {
```
Comment (Quy) [inline components/com_fields/layouts/fields/render.php]:
```suggestion
	if (!isset($content) || trim($content) == '') 
	{
```

**Comment by AndySDH** [comment]
Updated with @Quy suggestions, thanks!

File: components/com_fields/layouts/fields/render.php [hunk 3]
```diff
@@ -63,6 +63,12 @@
 	$layout = $field->params->get('layout', 'render');
 	$content = FieldsHelper::render($context, 'field.' . $layout, array('field' => $field));
 
+	// If the content is empty do nothing
+	if (!isset($content) || trim($content) == '')
```
Comment (Quy) [inline components/com_fields/layouts/fields/render.php]: `$content` is always set per line 64 so only need to test for empty string.

Comment (AndySDH) [reply components/com_fields/layouts/fields/render.php]: Makes sense

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on ff61169a1c2b882e0b53eb15ff82163e92a800da<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25145">issues.joomla.org/tracker/joomla-cms/25145</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on ff61169a1c2b882e0b53eb15ff82163e92a800da<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25145">issues.joomla.org/tracker/joomla-cms/25145</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25145">issues.joomla.org/tracker/joomla-cms/25145</a>.</sub>

File: components/com_fields/layouts/fields/render.php [hunk 4]
```diff
@@ -63,6 +63,12 @@
 	$layout = $field->params->get('layout', 'render');
 	$content = FieldsHelper::render($context, 'field.' . $layout, array('field' => $field));
 
+	// If the content is empty do nothing
+	if (trim($content) == '')
```
Comment (SharkyKZ) [inline components/com_fields/layouts/fields/render.php]:
```suggestion
	if (trim($content) === '') 
```

File: components/com_fields/layouts/fields/render.php [hunk 5]
```diff
@@ -54,7 +54,7 @@
 foreach ($fields as $field)
 {
 	// If the value is empty do nothing
-	if (!isset($field->value) || $field->value == '')
+	if (!isset($field->value) || trim($field->value) == '')
```
Comment (SharkyKZ) [inline components/com_fields/layouts/fields/render.php]:
```suggestion
	if (!isset($field->value) || trim($field->value) === '')
```

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on d6c16bed6110a0956d703ebaaaab19539f746933<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25145">issues.joomla.org/tracker/joomla-cms/25145</a>.</sub>

**Comment by AndySDH** [comment]
To merge in 3.9.8? 

Edit: 3.9.9 now I guess :D

**Comment by Quy** [comment]
@AndySDH Please apply the suggested changes. It will not require retesting.

**Comment by AndySDH** [comment]
I wasn't too sure the suggested changes made much sense as there's no need to compare for data type when we are comparing to an empty string.

But ok, I applied them.

**Comment by AndySDH** [comment]
Reminder about this for 3.9.9

**Comment by ghost** [comment]
> Reminder about this for 3.9.9

calling Release Lead @HLeithner to be sure.

**Comment by HLeithner** [comment]
thx

**Comment by ghost** [comment]
@AndySDH can you please mail me wohlkoenig.franz@gmail.com

**Comment by AndySDH** [comment]
> @AndySDH can you please mail me [wohlkoenig.franz@gmail.com](mailto:wohlkoenig.franz@gmail.com)

I did
